velero-plugins/configmap/restore.go

```diff
@@ -0,0 +1,39 @@
+package configmap
+
+import (
+	"github.com/konveyor/openshift-velero-plugin/velero-plugins/common"
+	"github.com/sirupsen/logrus"
+	"github.com/vmware-tanzu/velero/pkg/plugin/velero"
+	"k8s.io/apimachinery/pkg/api/meta"
+)
+
+// RestorePlugin is a restore item action plugin for Velero
+type RestorePlugin struct {
+	Log logrus.FieldLogger
+}
+
+// AppliesTo returns a velero.ResourceSelector that applies to configmaps
+func (p *RestorePlugin) AppliesTo() (velero.ResourceSelector, error) {
+	return velero.ResourceSelector{
+		IncludedResources: []string{"configmaps"},
+	}, nil
+}
+
+// Execute action for the restore plugin for the configmap resource
+// We want to exclude from backup configmaps that contain ownerReference to build pods
+func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*velero.RestoreItemActionExecuteOutput, error) {
+	p.Log.Info("[cm-restore] Entering ConfigMap restore plugin")
+	meta, err := meta.Accessor(input.Item)
+	if err != nil {
+		p.Log.Warnf("[cm-restore] Unable to accessed metadata, err: %v", err)
+	}
+	annotations := meta.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+	if _, ok := annotations[common.SkipBuildConfigConfigMapRestore]; ok {
```

Comment:
here we check if value exists only, right?

should we check if value is also true?

Reply:
Could do. I guess a mechanism for override? If already exists and false leave it as false.

Reply:
Is there a use case for overriding? If so, then yes, only set if not exists, and check for explicit true value. If we want to *always* skip, then we should leave it as-is. Also -- if we allow ConfigMap overriding, what about build pods? That one doesn't use annotations, so there's no way as-written to override. But I don't actually think we need that. Is there ever a case where not restoring build pod/configmap could break a restore? If there isn't one, then we don't need override.

Reply:
Note that for the various other scenarios where we already skip restore based on metadata, we don't have user override features either.

Reply:
yes check for the value as well. But maybe lets first identify a use case where you want to override this behavior and then add the override mechanism and doc it up.

Reply:
I don't think we have a use case other than "it doesn't hurt to have".. so might not be worth a doc.

Reply:
resolved in dff254445abd013a19e09781170358fe1f7fff37